### DISCUSSION TITLE
How to add && (and) in if condition with success or failure

### DISCUSSION THREAD

**Discussion by uditgaurav** [community/community#25381] [Actions]
<p>Hi all,<br>
I’m trying to run an action if it satisfies two conditions mentioned in if block:</p>
<pre><code class="lang-auto">      - name: Check for all the jobs are succeeded
        if: ${{ success() &amp;&amp;  $INITIAL_COUNT != $FINAL_COUNT }}
        uses: peter-evans/create-or-update-comment@v1
        with:
          comment-id: "${{ github.event.comment.id }}"
          body: |  
            **Test Result:** All tests are passed
          reactions: hooray
</code></pre>
<p>But this gives error:</p>
<pre><code class="lang-auto">The workflow is not valid. .github/workflows/main.yml (Line: 164, Col: 13): Unexpected symbol: '$INITIAL_COUNT'. Located at position 15 within expression: success() &amp;&amp;  $INITIAL_COUNT != $FINAL_COUNT
</code></pre>
<p>if I run only one condition i.e success one then it’s running fine.</p>
<p>This issue seems to be more of the <span class="math">`INITIAL_COUNT` &amp;&amp; </span><code>FINAL_COUNT</code> variables<br>
For this case also it is giving some error:</p>
<pre><code class="lang-auto">      - name: Check if any test ran or not
        if: $INITIAL_COUNT == $FINAL_COUNT
</code></pre>
<p>error:</p>
<pre><code class="lang-auto">The workflow is not valid. .github/workflows/main.yml (Line: 199, Col: 13): Unexpected symbol: '$INITIAL_COUNT'. Located at position 1 within expression: $INITIAL_COUNT == $FINAL_COUNT
</code></pre>
<p>So this is how these variables are set.</p>
<pre><code class="lang-auto">      - name: Record initial count
        env:
            RUN_INITIAL: ${{ github.run_number }}         
        run: echo ::set-env name=INITIAL_COUNT::$RUN_INITIAL

     
           ... some tasks ...

      - name: Record final count
        env:
            RUN_FINAL: ${{ github.run_number }}         
        run: echo ::set-env name=FINAL_COUNT::$RUN_FINAL
</code></pre>
<p>Can anyone help me to get the correct solution to this?<br>
Thanks!</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You set <code>INITIAL_COUNT</code> and <code>FINAL_COUNT</code> as environment variables, so in the <code>if:</code> clause you need to refer to them as <code>env.INITIAL_COUNT</code> and <code>env.FINAL_COUNT</code>, respectively. The <code>if</code> part isn’t a shell script like the <code>run</code> sections are, so you can’t use shell syntax for variables there.</p>
